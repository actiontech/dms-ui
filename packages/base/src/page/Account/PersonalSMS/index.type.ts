import { IGetUser } from '@actiontech/shared/lib/api/base/service/common';

export type FormFields = {
  enabled: boolean;
  code: string;
};

export type PersonalSMSProps = {
  userBaseInfo?: IGetUser;
  getUserInfo: () => void;
  loading: boolean;
};

export type ConfigFieldProps = {
  userPhone?: string;
  username?: string;
};
