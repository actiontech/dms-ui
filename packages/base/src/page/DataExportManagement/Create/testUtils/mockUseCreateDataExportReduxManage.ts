import { CreateDataExportPageEnum } from '../../../../store/dataExport';
import { ExportMethodEnum } from '../components/CreateTask/ExportMethodForm/index.enum';
import * as useCreateDataExportReduxManage from '../hooks/index.redux';

export const mockCreateDataExportRedux = {
  updateDataExportInfoOpen: false,
  formValues: {
    baseValues: { workflow_subject: 'test', desc: 'desc' },
    sourceValues: { dbService: '121' },
    methodValues: { sql: 'select 1;', exportMethod: ExportMethodEnum.sql }
  },
  pageState: CreateDataExportPageEnum.CREATE_TASK,
  auditLoading: false,
  submitLoading: false,
  taskIDs: ['11111'],
  workflowID: '123',
  updateFormValues: jest.fn(),
  updatePageState: jest.fn(),
  updateAuditLoading: jest.fn(),
  updateSubmitLoading: jest.fn(),
  initModalStatus: jest.fn(),
  updateModalStatus: jest.fn(),
  updateTaskIDs: jest.fn(),
  updateWorkflowID: jest.fn(),
  clearAllState: jest.fn()
};

export const mockUseCreateDataExportReduxManage = (
  data?: Partial<typeof mockCreateDataExportRedux>
) => {
  const spy = jest.spyOn(useCreateDataExportReduxManage, 'default');
  spy.mockImplementation(() => ({ ...mockCreateDataExportRedux, ...data }));
  return spy;
};
