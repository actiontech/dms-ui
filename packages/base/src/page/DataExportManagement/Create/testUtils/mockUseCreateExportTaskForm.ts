import * as useCreateExportTaskForm from '../hooks/useCreateExportTaskForm';

export const mockUseCreateExportTaskFormReturn = {
  baseForm: null as any,
  sourceForm: null as any,
  methodForm: null as any,
  auditLoading: false,
  formatSQLAction: jest.fn(),
  auditAction: jest.fn(),
  resetAllForms: jest.fn()
};

export const mockUseCreateExportTaskForm = (
  data?: Partial<typeof mockUseCreateExportTaskFormReturn>
) => {
  const spy = jest.spyOn(useCreateExportTaskForm, 'default');
  spy.mockImplementation(() => ({
    ...mockUseCreateExportTaskFormReturn,
    ...data
  }));
  return spy;
};
