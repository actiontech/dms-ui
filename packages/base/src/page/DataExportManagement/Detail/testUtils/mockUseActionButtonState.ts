import * as useActionButtonState from '../components/PageHeaderAction/useActionButtonState';

export const mockActionButtonStateData = {
  closeWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  approveWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  rejectWorkflowButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  },
  executeExportButtonMeta: {
    action: jest.fn(),
    loading: false,
    hidden: false
  }
};

export const mockUseActionButtonState = (
  data?: Partial<typeof mockActionButtonStateData>
) => {
  const spy = jest.spyOn(useActionButtonState, 'default');
  spy.mockImplementation(() => ({ ...mockActionButtonStateData, ...data }));
  return spy;
};
