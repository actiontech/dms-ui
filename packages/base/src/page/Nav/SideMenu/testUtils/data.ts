export const mockRecentlyOpenedProjectsData = {
  recentlyProjects: [
    {
      project_id: '1',
      project_name: 'test_project'
    }
  ],
  updateRecentlyProject: jest.fn(),
  currentProjectID: '0'
};
