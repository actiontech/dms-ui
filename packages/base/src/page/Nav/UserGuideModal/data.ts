import { GetUserSystemEnum } from '@actiontech/shared/lib/api/base/service/common.enum';
import { t } from '../../../locale';

export const systemDictionary = {
  [GetUserSystemEnum.WORKBENCH]: t('dmsMenu.userGuide.sqlWorkbench.label'),
  [GetUserSystemEnum.MANAGEMENT]: t('dmsMenu.userGuide.adminPanel.label')
};
