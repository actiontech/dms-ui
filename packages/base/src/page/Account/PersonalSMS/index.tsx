import { useTranslation } from 'react-i18next';
import { useBoolean } from 'ahooks';
import { useMemo, useEffect } from 'react';
import { Spin, message } from 'antd';
import { CustomLabelContent } from '@actiontech/shared/lib/components/CustomForm';
import ConfigField from './components/ConfigField';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import { FormFields } from './index.type';
import { switchFieldName } from './index.data';
import {
  useConfigRender,
  ConfigSwitch,
  ConfigSubmitButtonField,
  useConfigSwitchControls
} from '@actiontech/shared';
import { PersonalSMSProps } from './index.type';
import User from '@actiontech/shared/lib/api/base/service/User';
import SMS from '@actiontech/shared/lib/api/base/service/SMS';

const PersonalSMS: React.FC<PersonalSMSProps> = ({
  userBaseInfo,
  getUserInfo,
  loading
}) => {
  const { t } = useTranslation();

  const [messageApi, contextHolder] = message.useMessage();

  const [submitLoading, { setTrue: startSubmit, setFalse: submitFinish }] =
    useBoolean();

  const {
    form,
    renderConfigForm,
    startModify,
    modifyFinish,
    modifyFlag,
    enabled
  } = useConfigRender<FormFields>({
    switchFieldName,
    switchFieldLabel: (
      <CustomLabelContent title={t('dmsAccount.sms.title')} tips="" />
    )
  });

  useEffect(() => {
    if (!!userBaseInfo) {
      form.setFieldsValue({
        [switchFieldName]: !!userBaseInfo.two_factor_enabled
      });
    }
  }, [userBaseInfo, form]);

  const isConfigClosed = useMemo(() => {
    return !userBaseInfo?.two_factor_enabled;
  }, [userBaseInfo]);

  const handleClickModify = () => {
    startModify();
  };
  const handleClickCancel = () => {
    if (isConfigClosed) form.setFieldValue(switchFieldName, false);
    modifyFinish();
  };

  const onSubmit = async (isCodingEnabled: boolean, values?: FormFields) => {
    startSubmit();
    if (isCodingEnabled) {
      const verificationRes = await SMS.VerifySmsCode({
        code: values?.code,
        username: userBaseInfo?.name
      });
      if (verificationRes.data.data?.verify_error_message) {
        messageApi.error(verificationRes.data.data?.verify_error_message);
      }
      if (
        verificationRes.data.code !== ResponseCode.SUCCESS ||
        !verificationRes.data.data?.is_verify_sent_normally
      ) {
        submitFinish();
        return;
      }
    }
    User.UpdateCurrentUser({
      current_user: {
        two_factor_enabled: isCodingEnabled
      }
    })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          messageApi.success(t('dmsAccount.sms.updateSuccessTips'));
          modifyFinish();
          form.resetFields();
          getUserInfo();
        }
      })
      .finally(() => {
        submitFinish();
      });
  };

  const onConfigSwitchPopoverConfirm = () => {
    if (!userBaseInfo?.two_factor_enabled && modifyFlag) {
      handleClickCancel();
      hiddenConfigSwitchPopover();
    } else {
      onSubmit(false);
    }
  };

  const {
    configSwitchPopoverOpenState,
    generateConfigSwitchPopoverTitle,
    onConfigSwitchPopoverOpen,
    handleConfigSwitchChange,
    hiddenConfigSwitchPopover
  } = useConfigSwitchControls(form, switchFieldName);

  return (
    <div>
      {contextHolder}
      <Spin spinning={loading}>
        {renderConfigForm({
          data: userBaseInfo ?? {},
          columns: [],
          configExtraButtons: null,
          configSwitchNode: (
            <ConfigSwitch
              title={generateConfigSwitchPopoverTitle(modifyFlag)}
              switchFieldName={switchFieldName}
              submitLoading={submitLoading}
              popoverVisible={configSwitchPopoverOpenState}
              onConfirm={onConfigSwitchPopoverConfirm}
              onSwitchChange={(open) => {
                if (open && !userBaseInfo?.phone) {
                  messageApi.error(t('dmsAccount.sms.noPhoneNumbersTips'));
                  form.setFieldValue(switchFieldName, false);
                  return;
                }
                handleConfigSwitchChange(open, handleClickModify);
              }}
              onSwitchPopoverOpen={onConfigSwitchPopoverOpen}
            />
          ),
          configField: (
            <ConfigField
              userPhone={userBaseInfo?.phone}
              username={userBaseInfo?.name}
            />
          ),
          submitButtonField: (
            <ConfigSubmitButtonField
              submitLoading={submitLoading}
              handleClickCancel={handleClickCancel}
            />
          ),
          onSubmit: (values) => {
            onSubmit(enabled, values);
          }
        })}
      </Spin>
    </div>
  );
};

export default PersonalSMS;
