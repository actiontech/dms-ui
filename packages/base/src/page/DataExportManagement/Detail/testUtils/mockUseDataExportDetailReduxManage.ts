import { GetDataExportTaskStatusEnum } from '@actiontech/shared/lib/api/base/service/common.enum';
import * as useDataExportDetailReduxManage from '../hooks/index.redux';
import { GetDataExportWorkflowResponseData } from '../../../../testUtils/mockApi/dataExport/data';

export const mockDataExportDetailRedux = {
  workflowStepOpen: false,
  workflowRejectOpen: false,
  workflowInfo: GetDataExportWorkflowResponseData,
  taskInfos: [
    {
      task_uid: '1752172791873933312',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.finish,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933312.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    },
    {
      task_uid: '1752172791873933313',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.exporting,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933313.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    },
    {
      task_uid: '1752172791873933314',
      db_info: {
        uid: '1752170583749038080',
        name: 'mysql-1',
        db_type: '',
        database_name: 'test'
      },
      status: GetDataExportTaskStatusEnum.failed,
      export_start_time: '2024-01-30T11:32:15.085+08:00',
      export_end_time: '2024-01-30T11:32:15.085+08:00',
      file_name: 'mysql-1_20240130113114-1752172791873933314.zip',
      audit_result: {
        audit_level: '',
        score: 100,
        pass_rate: 1
      },
      export_type: 'SQL',
      export_file_type: 'CSV'
    }
  ],
  curTaskID: '1752172791873933312',
  taskStatusNumber: {
    success: 1,
    failed: 0,
    exporting: 0
  },
  canRejectWorkflow: false,
  updateWorkflowStepOpen: jest.fn(),
  updateWorkflowRejectOpen: jest.fn(),
  updateWorkflowInfo: jest.fn(),
  updateTaskInfos: jest.fn(),
  updateCurTaskID: jest.fn(),
  updateTaskStatusNumber: jest.fn(),
  updateCanRejectWorkflow: jest.fn(),
  clearAllDetailState: jest.fn()
};

export const mockUseDataExportDetailReduxManage = (
  data?: Partial<typeof mockDataExportDetailRedux>
) => {
  const spy = jest.spyOn(useDataExportDetailReduxManage, 'default');
  spy.mockImplementation(() => ({ ...mockDataExportDetailRedux, ...data }));
  return spy;
};
