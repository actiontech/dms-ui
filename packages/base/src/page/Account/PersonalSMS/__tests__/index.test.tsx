import sms from '../../../../testUtils/mockApi/sms';
import { cleanup, fireEvent, act, screen } from '@testing-library/react';
import { getBySelector } from '@actiontech/shared/lib/testUtil/customQuery';
import PersonalSMS from '..';
import { createSpySuccessResponse } from '@actiontech/shared/lib/testUtil/mockApi';
import { mockUseCurrentUser } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentUser';
import { mockUserInfo } from '@actiontech/shared/lib/testUtil/mockHook/data';
import account from '../../../../testUtils/mockApi/account';
import { baseSuperRender } from '../../../../testUtils/superRender';

describe('base/System/GlobalSetting/PersonalSMS', () => {
  let updateCurrentUserSpy: jest.SpyInstance;
  let sendSmsCodeSpy: jest.SpyInstance;
  let verifySmsCodeSpy: jest.SpyInstance;
  let getUserInfoSpy = jest.fn();
  const customRender = ({
    phone = mockUserInfo.userInfo.phone,
    enabled = true
  }) => {
    return baseSuperRender(
      <PersonalSMS
        userBaseInfo={{
          ...mockUserInfo.userInfo,
          two_factor_enabled: enabled,
          phone: phone
        }}
        getUserInfo={getUserInfoSpy}
        loading={false}
      />
    );
  };

  beforeEach(() => {
    mockUseCurrentUser();
    jest.useFakeTimers();
    updateCurrentUserSpy = account.updateCurrentUser();
    sendSmsCodeSpy = sms.sendSmsCode();
    verifySmsCodeSpy = sms.verifySmsCode();
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    cleanup();
  });

  it('render init snap', async () => {
    const { baseElement } = customRender({});
    await act(async () => jest.advanceTimersByTime(0));
    expect(baseElement).toMatchSnapshot();
    expect(getBySelector('#enabled')).toBeChecked();
  });

  it('render close personal sms', async () => {
    customRender({});
    await act(async () => jest.advanceTimersByTime(3300));
    fireEvent.click(getBySelector('#enabled'));
    await act(async () => jest.advanceTimersByTime(500));
    expect(screen.getByText('是否确认关闭当前配置？')).toBeInTheDocument();

    fireEvent.click(screen.getByText('确 认'));
    await act(async () => jest.advanceTimersByTime(0));
    expect(updateCurrentUserSpy).toHaveBeenCalledTimes(1);
    expect(updateCurrentUserSpy).toHaveBeenCalledWith({
      current_user: {
        two_factor_enabled: false
      }
    });
    await act(async () => jest.advanceTimersByTime(3000));
    expect(getUserInfoSpy).toHaveBeenCalledTimes(1);
  });

  it('render open personal sms', async () => {
    customRender({ enabled: false });
    await act(async () => jest.advanceTimersByTime(3000));
    const switchEle = getBySelector('#enabled');
    expect(switchEle).not.toBeChecked();

    fireEvent.click(switchEle);
    await act(async () => jest.advanceTimersByTime(300));

    fireEvent.click(screen.getByText('发送验证码'));
    await act(async () => jest.advanceTimersByTime(3000));
    expect(sendSmsCodeSpy).toHaveBeenCalledTimes(1);

    fireEvent.change(getBySelector('#code'), {
      target: {
        value: '1234'
      }
    });
    await act(async () => jest.advanceTimersByTime(0));

    fireEvent.click(screen.getByText('提 交'));
    await act(async () => jest.advanceTimersByTime(0));
    expect(verifySmsCodeSpy).toHaveBeenCalledTimes(1);
    expect(verifySmsCodeSpy).toHaveBeenNthCalledWith(1, {
      code: '1234',
      username: mockUserInfo.userInfo.name
    });
    await act(async () => jest.advanceTimersByTime(3000));
    expect(updateCurrentUserSpy).toHaveBeenCalledTimes(1);
    expect(updateCurrentUserSpy).toHaveBeenCalledWith({
      current_user: {
        two_factor_enabled: true
      }
    });
    await act(async () => jest.advanceTimersByTime(3000));
    expect(getUserInfoSpy).toHaveBeenCalledTimes(1);
  });

  it('render open personal sms when user phone is empty string', async () => {
    customRender({ enabled: false, phone: '' });
    await act(async () => jest.advanceTimersByTime(3000));
    const switchEle = getBySelector('#enabled');
    expect(switchEle).not.toBeChecked();

    fireEvent.click(switchEle);
    await act(async () => jest.advanceTimersByTime(0));

    expect(
      screen.getByText('请先完成手机号绑定后开启双因素认证')
    ).toBeInTheDocument();
  });

  it('render open personal sms when verify sms code fail', async () => {
    verifySmsCodeSpy.mockImplementation(() =>
      createSpySuccessResponse({
        data: {
          is_verify_sent_normally: false,
          verify_error_message: '验证码错误'
        }
      })
    );
    customRender({ enabled: false });
    await act(async () => jest.advanceTimersByTime(3000));
    const switchEle = getBySelector('#enabled');
    expect(switchEle).not.toBeChecked();

    fireEvent.click(switchEle);
    await act(async () => jest.advanceTimersByTime(300));

    fireEvent.click(screen.getByText('发送验证码'));
    await act(async () => jest.advanceTimersByTime(3000));
    expect(sendSmsCodeSpy).toHaveBeenCalledTimes(1);

    fireEvent.change(getBySelector('#code'), {
      target: {
        value: '1234'
      }
    });
    await act(async () => jest.advanceTimersByTime(0));

    fireEvent.click(screen.getByText('提 交'));
    await act(async () => jest.advanceTimersByTime(0));
    expect(verifySmsCodeSpy).toHaveBeenCalledTimes(1);
    expect(verifySmsCodeSpy).toHaveBeenNthCalledWith(1, {
      code: '1234',
      username: mockUserInfo.userInfo.name
    });
    await act(async () => jest.advanceTimersByTime(3000));
    expect(screen.getByText('验证码错误')).toBeInTheDocument();
    await act(async () => jest.advanceTimersByTime(3000));
    expect(updateCurrentUserSpy).not.toHaveBeenCalled();
  });
});
