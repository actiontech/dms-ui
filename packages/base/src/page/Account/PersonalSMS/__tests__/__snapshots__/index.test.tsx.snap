// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base/System/GlobalSetting/PersonalSMS render init snap 1`] = `
<body>
  <div>
    <div>
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <section
            class="css-zupqof"
          >
            <form
              class="ant-form ant-form-horizontal"
            >
              <div
                class="switch-field-wrapper"
              >
                <div
                  class="ant-form-item css-1jlm9cy"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-col-11 ant-form-item-label ant-form-item-label-left"
                    >
                      <label
                        class="ant-form-item-no-colon"
                        title="双因素认证"
                      >
                        双因素认证
                      </label>
                    </div>
                    <div
                      class="ant-col ant-col-11 ant-col-push-2 ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                            >
                              <div
                                class="ant-form-item css-1jlm9cy ant-form-item-has-success"
                              >
                                <div
                                  class="ant-row ant-form-item-row"
                                >
                                  <div
                                    class="ant-col ant-form-item-control"
                                  >
                                    <div
                                      class="ant-form-item-control-input"
                                    >
                                      <div
                                        class="ant-form-item-control-input-content"
                                      >
                                        <button
                                          aria-checked="true"
                                          class="ant-switch basic-switch-wrapper system-config-switch css-g6dhbn ant-switch-checked"
                                          id="enabled"
                                          role="switch"
                                          type="button"
                                        >
                                          <div
                                            class="ant-switch-handle"
                                          />
                                          <span
                                            class="ant-switch-inner"
                                          >
                                            <span
                                              class="ant-switch-inner-checked"
                                            />
                                            <span
                                              class="ant-switch-inner-unchecked"
                                            />
                                          </span>
                                        </button>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="config-field-wrapper"
                hidden=""
              >
                <div
                  class="ant-form-item css-1jlm9cy"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-col-11 ant-form-item-label ant-form-item-label-left"
                    >
                      <label
                        class="ant-form-item-no-colon"
                        title="手机号"
                      >
                        手机号
                      </label>
                    </div>
                    <div
                      class="ant-col ant-col-11 ant-col-push-2 ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          13112341234
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item css-1jlm9cy"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-col-11 ant-form-item-label ant-form-item-label-left"
                    >
                      <label
                        class="ant-form-item-required ant-form-item-no-colon"
                        for="code"
                        title="验证码"
                      >
                        验证码
                      </label>
                    </div>
                    <div
                      class="ant-col ant-col-11 ant-col-push-2 ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center css-jxd35n"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <input
                                aria-required="true"
                                class="ant-input basic-input-wrapper css-1pd1cd5"
                                id="code"
                                placeholder="请输入验证码"
                                type="text"
                                value=""
                              />
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn ant-btn-default basic-button-wrapper css-1uxy5k6"
                                type="button"
                              >
                                <span>
                                  发送验证码
                                </span>
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item css-15d6abf"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-col-11 ant-form-item-label ant-form-item-label-left"
                    >
                      <label
                        class="ant-form-item-no-colon"
                        title=" "
                      >
                         
                      </label>
                    </div>
                    <div
                      class="ant-col ant-col-11 ant-col-push-2 ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 12px;"
                            >
                              <button
                                class="ant-btn ant-btn-default basic-button-wrapper css-1uxy5k6"
                                type="button"
                              >
                                <span>
                                  取 消
                                </span>
                              </button>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn ant-btn-primary basic-button-wrapper css-1uxy5k6"
                                type="submit"
                              >
                                <span>
                                  提 交
                                </span>
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </section>
        </div>
      </div>
    </div>
  </div>
</body>
`;
